pkg/sign/certificate.go

```diff
@@ -42,6 +46,8 @@ type FulcioOptions struct {
 	Retries uint
 	// Optional version string for user agent
 	LibraryVersion string
+	// Optional HTTP client (for dependency injection)
+	Client HTTPClient
```

Comment:
Nit: I suggest using [`http.RoundTripper`](https://pkg.go.dev/net/http#RoundTripper) for dependency injection.

Reply:
I was reading a bit about the idiomatic way to do testing in Golang, and a common [suggestion](https://medium.com/zus-health/mocking-outbound-http-requests-in-go-youre-probably-doing-it-wrong-60373a38d2aa#:~:text=There%20is%20an%20easier%2C%20idiomatic%20way) was https://pkg.go.dev/net/http/httptest

Reply:
Interesting, I've used the pkg httptest for testing handlers before, but not outbound requests. I think for this usecase, I still prefer replacing the transport with a custom `RoundTripper` as it seems simpler and doesn't open a port in tests.